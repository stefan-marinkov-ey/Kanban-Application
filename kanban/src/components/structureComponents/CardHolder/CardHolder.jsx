import React from "react";
import { threeDots } from "../../../utility/constantsText";
import { makeLighterColor } from "../../../utility/utilityFunc";
import Modal from "../Modal/Modal";
import EditCard from "../EditCard/EditCard";
import useModal from "../Modal/useModal";
import CardDescription from "../CardDescription/CardDescription";

const CardHolder = ({
  cardName,
  cardColor,
  cardDesc,
  cardId,
  refreshCard,
  setRefreshCard,
}) => {
  const { isShowing, toggle } = useModal();
  const { isShowing: Desc, toggle: toggleDesc } = useModal();
  const elementP = makeLighterColor(cardColor);

  return (
    <div className="cardFunctionality">
      <p onClick={toggle} style={elementP}>
        {cardName} <span>{threeDots}</span>
      </p>
      <p onClick={toggleDesc} style={elementP}>
        Click for description
      </p>
      <Modal isShowing={Desc} hide={toggleDesc}>
        <CardDescription
          cardDesc={cardDesc}
          cardName={cardName}
          elementP={elementP}
          toggleDesc={toggleDesc}
          Desc={Desc}
        />
      </Modal>
      <Modal isShowing={isShowing} hide={toggle}>
        <EditCard
          cardId={cardId}
          cardDesc={cardDesc}
          cardColor={cardColor}
          cardName={cardName}
          refreshCard={refreshCard}
          setRefreshCard={setRefreshCard}
          toggle={toggle}
        />
      </Modal>
    </div>
  );
};

export default CardHolder;
