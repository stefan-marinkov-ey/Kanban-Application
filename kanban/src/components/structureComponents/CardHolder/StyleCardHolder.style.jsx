import styled from "styled-components";

export const StyleCardHolder = styled.div`
  width: 85%;
  color: ${(props) => props.theme.primaryText};
  .cardFunc {
    width: 95%;
    height: 2rem;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 0 5%;
    font-weight: bolder;
    box-shadow: 0.5px 0.5px 0.5px 0.5px ${(props) => props.theme.primaryText};
    span {
      visibility: hidden;
    }
  }
  .cardDesc {
    width: 95%;
    height: 2rem;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
    font-size: smaller;
    display: flex;
    align-items: center;
    justify-content: center;
  }
  p:hover {
    cursor: pointer;
    opacity: 0.8;
    background: ${(props) => props.theme.backgroundColor};
    transition: all 0.5s ease-in;
    span {
      visibility: visible;
    }
    transition: all 0.5s ease-in;
  }
`;
