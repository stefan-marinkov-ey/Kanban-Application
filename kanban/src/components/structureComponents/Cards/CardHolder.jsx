import React from "react";
import { threeDots } from "../../../utility/constantsText";
import { makeLighterColor } from "../../../utility/utilityFunc";

const CardHolder = ({ cardName, cardColor }) => {
  const elementP = makeLighterColor(cardColor);

  return (
    <div className="cardFunctionality">
      <p onClick={console.log("Placeholder For Editing Card")} style={elementP}>
        {cardName} <span>{threeDots}</span>
      </p>
    </div>
  );
};

export default CardHolder;
