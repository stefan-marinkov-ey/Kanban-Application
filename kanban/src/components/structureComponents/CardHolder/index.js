import CardHolder from "./CardHolder";

export default CardHolder;
export { CardHolder };
