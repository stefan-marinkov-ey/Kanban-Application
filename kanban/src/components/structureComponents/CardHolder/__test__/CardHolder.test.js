import React from "react";
import { shallow } from "enzyme";
import CardHolder from "../CardHolder";

describe("Holder", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CardHolder />);
  });
  it("render component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
