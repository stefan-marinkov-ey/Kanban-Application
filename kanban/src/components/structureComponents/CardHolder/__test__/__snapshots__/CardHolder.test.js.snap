// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Holder render component 1`] = `ShallowWrapper {}`;
